src/OpenFeature/Providers/Memory/InMemoryProvider.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using OpenFeature.Constant;
+using OpenFeature.Error;
+using OpenFeature.Model;
+
+namespace OpenFeature.Providers.Memory
+{
+    /// <summary>
+    /// The in memory provider.
+    /// Useful for testing and demonstration purposes.
+    /// </summary>
+    /// <seealso href="https://openfeature.dev/specification/appendix-a#in-memory-provider">In Memory Provider specification</seealso>
+    public class InMemoryProvider : FeatureProvider
```

Comment:
```suggestion
    public sealed class InMemoryProvider : FeatureProvider
```

Reply:
I think I'd rather not seal this class unless you have a good reason for us to...

I've extended in-memory providers to create artificial delays for our React SDK testing (to test suspense), extended it to support "fake" events, etc. I think it's nice to be able to do this.

Reply:
Agreed! That's a valid point.